src/libs/sso-utils.js

```diff
@@ -218,23 +218,29 @@ export const checkSSOGroup = async (credentials, userId, targetGroups = []) => {
  */
 export const checkUserAuthStatus = async userInfo => {
   const accountStatus = { isPending: false, isAuthorized: false, isRejected: false };
-  // User need to have a valid profile before they become authorized:
-  if (!checkUserProfile(userInfo)) return accountStatus;
 
   try {
-    const matchSchema = await checkRocketChatSchema(userInfo);
-    // if user account is not matching the requirments, reject:
-    if (!matchSchema) return { ...accountStatus, ...{ isRejected: true } };
     const ssoGroupNames = userInfo.group.map(i => i.name);
+    const isPending = ssoGroupNames.includes('pending');
+    const isAuthorized = ssoGroupNames.includes('registered');
+
     // if user has complete profile and matches requirement, return the current sso group status:
-    return {
-      ...accountStatus,
-      ...{
-        isPending: ssoGroupNames.includes('pending'),
-        isAuthorized: ssoGroupNames.includes('registered'),
-      },
-    };
-    // TODO: add check on authorization by email invitation token
+    if (isPending || isAuthorized) {
+      // User need to have a valid profile before they become authorized:
+      if (!checkUserProfile(userInfo)) return accountStatus;
+      return {
+        ...accountStatus,
```

Comment:
since this function doesn't mutate any object. You don't need to spread everything. You can declare/set the object key/value pairs as needed

Reply:
thought about it. Will still keep it for now, cuz I might need more status into the `accoutStatus` soon. Might be better to spread it instead of repeating all the ones separately...